python/ray/serve/schema.py

```diff
@@ -287,6 +285,13 @@ def _deployment_info_to_schema(name: str, info: DeploymentInfo) -> DeploymentSch
         is_driver_deployment=info.is_driver_deployment,
     )
 
+    if info.deployment_config.autoscaling_config is not None:
```

Comment:
if DeploymentSchema is a pydantic model, should we update the pydantic validator? why we are adding this logic here

Reply:
This is a user-facing model for specifying config options for a deployment. The two places the check used currently is:
- when a user submits a config to Serve
- returned here as a copy of the config the user submitted, except with default values all filled in
I think the validator makes sense in these 2 cases.

Reply:
i think standard practice in these cases is to have separate create models. Eg. you can do 
```
class DeploymentConfig(BaseModel):
    ...

class CreateDeploymentConfig(DeploymentConfig):
    # differences for the create model
```
generally speaking it's not good practice to use the same model for create / get paths for this class of reasons.

Reply:
Got it, that makes sense. I've created a separate issue to track this: https://github.com/ray-project/ray/issues/34383.